pkg/deployment/note/show.go

```diff
@@ -21,24 +21,28 @@ import (
 	"github.com/elastic/cloud-sdk-go/pkg/api"
 	"github.com/elastic/cloud-sdk-go/pkg/client/deployments"
 	"github.com/elastic/cloud-sdk-go/pkg/models"
+
+	"github.com/elastic/ecctl/pkg/deployment"
 )
 
-// List lists all of the notes for the deployment
-func List(params ListParams) (*models.Notes, error) {
-	if err := params.Validate(); err != nil {
-		return nil, err
-	}
+// GetParams is used on List
+type GetParams struct {
+	Params
+}
 
-	res, err := params.API.V1API.Deployments.GetDeploymentNotes(
-		deployments.NewGetDeploymentNotesParams().
-			WithDeploymentID(params.ID),
-		params.AuthWriter,
-	)
+// Use different resource types when this is supported by the API.
+// For the time being, the notes endpoint only allows elasticsearch IDs.
+func (params *GetParams) fillDefaults() error {
```

Comment:
we're doing this same workflow in a few places. I think it'd be better to tie the fillDefaults to a `NoteParams` and then embed that or just use it wherever needed.